src/presenter.rs

```diff
@@ -104,6 +115,11 @@ impl<'a> Presenter<'a> {
             self.render(&mut drawer)?;
 
             loop {
+                if let Some(idx) = self.state.presentation().listen_for_speaker_note_evt() {
```

Comment:
I think it would be interesting for this thing to be inside `input/source.rs` and act as a normal command, e.g. it could return `Command::GoToSlide` when such an event is found. This would make it more transparent to the presenter.

Reply:
That would make sense - it _is_ a source of a command after all. 

So are you thinking:

- Remove the `SpeakerNoteChannel` enum,
- Store the `Option<iceoryx2::Listener>` directly in `CommandSource` if viewing speaker notes,
- Store the `Option<iceoryx2::Notifier>` directly in `Presenter` for the main presentation, if presenting with speaker notes.
?

Reply:
Yes, exactly :ok_hand:

Reply:
Done.